src/daft-functions/src/distance/cosine.rs

```diff
@@ -57,33 +60,31 @@ impl ScalarUDF for CosineDistanceFunction {
         } = inputs.try_into()?;
         let source = source.to_field(schema)?;
         let query = query.to_field(schema)?;
-        let source_is_numeric = source.dtype.is_fixed_size_numeric();
-        let query_is_numeric = query.dtype.is_fixed_size_numeric();
 
-        if let Ok((source_size, query_size)) = source
-            .dtype
-            .fixed_size()
-            .and_then(|source| query.dtype.fixed_size().map(|q| (source, q)))
-        {
-            if source_size != query_size {
-                return Err(DaftError::ValueError(format!(
-                    "Expected source and query to have the same size, instead got {source_size} and {query_size}"
-                )));
+        match (&source.dtype.to_physical(), &query.dtype.to_physical()) {
+            (
+                DataType::FixedSizeList(source_inner_dtype, source_size),
+                DataType::FixedSizeList(query_inner_dtype, query_size),
+            ) => {
+                if source_inner_dtype != query_inner_dtype {
+                    return Err(DaftError::ValueError(format!(
+                        "Expected source and query to have the same inner dtype, instead got {} and {}",
+                        source_inner_dtype, query_inner_dtype
+                    )));
```

Comment:
Check out `value_error!` in common::errors. I've been trying to use this as well as using macros for errors in case we ever want to add more context to where errors occur.